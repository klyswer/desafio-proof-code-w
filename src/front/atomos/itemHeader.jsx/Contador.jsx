import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Contador = ({number}) => {
  return (
    <div className="cantidad_productos">
      {number}
    </div>
  );
}

Contador.defaultProps = {
  number:'0',
}

Contador.propTypes = {
  number:PropTypes.number
};

export default Contador;
