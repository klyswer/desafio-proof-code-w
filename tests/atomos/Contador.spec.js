import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import Contador from '../../src/front/atomos/itemHeader/Contador';

describe('<Contador />', () => {

  const numberTest = 5;
  const wrapper = shallow( <Contador number={numberTest}  /> );


  test('Primer SnapShot: <Contador />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Caso prueba 1: validando render de props number:number.', () => {
    const numeroEsperado = wrapper.find('.cantidad_productos').text();
    expect(numeroEsperado).toEqual(numberTest.toString());
  });

});