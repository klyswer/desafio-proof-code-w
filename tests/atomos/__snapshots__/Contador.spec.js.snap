// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contador /> Primer SnapShot: <Contador /> 1`] = `
<div
  className="cantidad_productos"
>
  5
</div>
`;
